modules/tinymce/src/plugins/table/main/ts/ui/UiUtils.ts

```diff
@@ -14,12 +14,12 @@ interface Item {
   readonly value: string;
 }
 
-const onSetupToggle = (editor: Editor, formatName: string, formatValue: string) => {
+const onSetupToggle = (editor: Editor, formatName: string, active: () => boolean) => {
   return (api: Toolbar.ToolbarMenuButtonInstanceApi) => {
     const boundCallback = Singleton.unbindable();
 
     const init = () => {
-      api.setActive(editor.formatter.match(formatName, { value: formatValue }));
+      api.setActive(active());
       const binding = editor.formatter.formatChanged(formatName, api.setActive);
```

Comment:
This change doesn't handle the `formatChanged` case, as that needs to call `api.setActive(true)` when nothing matches as well, as the above call only handles the very first init and doesn't handle cases after that.

Worth noting `formatChanged` has a `similar` argument that you could pass to be `true` when the format value is empty. You can probably then check if that returns false, then it means nothing that uses the same format with a different value matches. If not, you might be able to use your `active` callback here, however I do have concerns about the performance impacts of that, as it needs to check every value is not set. What we nearly need to be able to do this in a performant way is to be able to pass `similar` to the match function as well.